proxy/proxy_script.js

```diff
@@ -0,0 +1,18 @@
+// Keep track of any local settings for the addon.
+let settings = {
+  enabled: false,
+};
+
+// Required PAC function that will be called to determine
+// if a proxy should be used.
+function FindProxyForURL(url, host) {
+  if (settings.enabled) {
+    return "DIRECT 1.2.3.4:8080"; // Update to return proxy server rules (e.g. "PROXY 1.2.3.4:8080")
+  }
+  return "DIRECT 1.2.3.4:8080";
+}
+
+// Listen for messages from the background script.
+browser.runtime.onMessage.addListener(msg => {
+  settings.enabled = msg.enabled;
```

Comment:
This gets sent by the background script right when the extension is installed. The main purpose for this is to show that the proxy script supports runtime.onMessage.